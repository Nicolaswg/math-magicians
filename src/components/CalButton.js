import React from 'react';
import PropTypes from 'prop-types';

const Buttons = ({ click }) => {
  const values = [
    'AC',
    '+/-',
    '%',
    'รท',
    '7',
    '8',
    '9',
    'x',
    '4',
    '5',
    '6',
    '-',
    '1',
    '2',
    '3',
    '+',
    '0',
    '.',
    '=',
  ];
  return (
    <div className="row">
      {values.map((e) => {
        if (e === '0') {
          return (
            <button type="button" className="button span-2" key={e} onClick={click}>{e}</button>
          );
        }
        return (
          <button type="button" className="button" key={e} onClick={click}>{e}</button>
        );
      })}
    </div>
  );
};

Buttons.propTypes = {
  click: PropTypes.func.isRequired,
};

export default Buttons;
