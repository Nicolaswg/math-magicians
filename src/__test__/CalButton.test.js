import renderer from 'react-test-renderer';
import CalButton from '../components/CalButton';

it('Render Calculator correctly', () => {
  const tree = renderer.create(
    <CalButton />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
