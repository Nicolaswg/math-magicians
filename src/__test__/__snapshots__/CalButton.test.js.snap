// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Calculator correctly 1`] = `
<div
  className="row"
>
  <button
    className="button"
    type="button"
  >
    AC
  </button>
  <button
    className="button"
    type="button"
  >
    +/-
  </button>
  <button
    className="button"
    type="button"
  >
    %
  </button>
  <button
    className="button"
    type="button"
  >
    รท
  </button>
  <button
    className="button"
    type="button"
  >
    7
  </button>
  <button
    className="button"
    type="button"
  >
    8
  </button>
  <button
    className="button"
    type="button"
  >
    9
  </button>
  <button
    className="button"
    type="button"
  >
    x
  </button>
  <button
    className="button"
    type="button"
  >
    4
  </button>
  <button
    className="button"
    type="button"
  >
    5
  </button>
  <button
    className="button"
    type="button"
  >
    6
  </button>
  <button
    className="button"
    type="button"
  >
    -
  </button>
  <button
    className="button"
    type="button"
  >
    1
  </button>
  <button
    className="button"
    type="button"
  >
    2
  </button>
  <button
    className="button"
    type="button"
  >
    3
  </button>
  <button
    className="button"
    type="button"
  >
    +
  </button>
  <button
    className="button span-2"
    type="button"
  >
    0
  </button>
  <button
    className="button"
    type="button"
  >
    .
  </button>
  <button
    className="button"
    type="button"
  >
    =
  </button>
</div>
`;
